INTERNSHIP/env/Lib/site-packages/pip/_internal/index/package_finder.py

```
@@ -0,0 +1,1027 @@
+        if installed_version is None and best_candidate is None:
+            logger.critical(
+                "Could not find a version that satisfies the requirement %s "
+                "(from versions: %s)",
+                req,
+                _format_versions(best_candidate_result.iter_all()),
+            )
+
+            raise DistributionNotFound(f"No matching distribution found for {req}")
+
+        def _should_install_candidate(
+            candidate: Optional[InstallationCandidate],
```

Ruff (0.12.2)

Unused function argument: `candidate`

(ARG001)